examples/dtls-sock/server_keys.h

```diff
@@ -0,0 +1,60 @@
+#ifndef SERVER_KEYS_H
+#define SERVER_KEYS_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ *  Default keys examples for tinyDTLS (for RIOT, Linux and Contiki)
+ */
+#ifdef DTLS_PSK
+#define PSK_DEFAULT_IDENTITY "Client_identity"
+#define PSK_DEFAULT_KEY "secretPSK"
+#define PSK_OPTIONS "i:k:"
+#define PSK_ID_MAXLEN 32
+#define PSK_MAXLEN 32
+
+#endif /* DTLS_PSK */
+
+#ifdef DTLS_ECC
+static const unsigned char server_ecdsa_priv_key[] = {
```

Comment:
`static const` in a header will duplicate the memory for every inclusion of the header. Rather use `#define`s here, or define a `common.c` where you can return a pointer to these arrays with a function.

Reply:
`#define`s might be too complicated for a multiline variable like this with the line continuation character. If I understands you correctly, with the `common.c` idea the user will have to write more code to just pass the certs.

How about doing it like how the wolfSSL PR does it [here][1]. Then in the .c file where the certs will be used, these declarations will be needed:

```C
extern const unsigned char server_cert[];
extern const unsigned long server_cert_len;
```

[1]: https://github.com/RIOT-OS/RIOT/pull/10308/files#diff-09ce13303248a398e36e3febb371129a

Reply:
> `#define`s might be too complicated for a multiline variable like this with the line continuation character.

Why would it be more complicated? You just need to add `\`s.

```C
#define SERVER_ECDSA_PRIV_KEY { \
    0xa3, 0x8e, 0x19, 0xd9, 0x25, 0xd8, 0xf4, 0xf4, \
    0x96, 0x47, 0x3e, 0x4f, 0xd4, 0x6a, 0x12, 0xb2, \
    0xf5, 0xa1, 0xb5, 0x06, 0xa2, 0x42, 0x92, 0x3c, \
    0xfd, 0xf5, 0x4c, 0x6d, 0x01, 0x75, 0x89, 0x6f, \
}
```

> How about doing it like how the wolfSSL PR does it [here](https://github.com/RIOT-OS/RIOT/pull/10308/files#diff-09ce13303248a398e36e3febb371129a). Then in the .c file where the certs will be used, these declarations will be needed:
> 
> ```c
> extern const unsigned char server_cert[];
> extern const unsigned long server_cert_len;
> ```

As var as I understand `const` instead of `static const` comes with some memory caveats, especially when assigning the variable, but I would need to read up on that myself again.

Reply:
> Why would it be more complicated? You just need to add `\`s.

That will need the user to change the file themselves.

> How about doing it like how the wolfSSL PR does it [here](https://github.com/RIOT-OS/RIOT/pull/10308/files#diff-09ce13303248a398e36e3febb371129a). Then in the .c file where the certs will be used, these declarations will be needed:
> 
> ```c
> extern const unsigned char server_cert[];
> extern const unsigned long server_cert_len;
> ```

This uses certificate file converted to hex in a .c file using `xxd -i`, so no more changes to the file are needed.

Reply:
> > Why would it be more complicated? You just need to add `\`s.
> 
> That will need the user to change the file themselves.

The current approach does so as well.

> This uses certificate file converted to hex in a .c file using `xxd -i`, so no more changes to the file are needed.

This = "the approach here" or the WolfSSL approach?

Reply:
See https://github.com/RIOT-OS/RIOT/pull/11943#issuecomment-534987491